src/query.md

```diff
@@ -260,40 +259,13 @@ impl<'tcx> QueryConfig for type_of<'tcx> {
 }
 ```
 
-There is an additional trait that you may wish to implement called
-[`self::config::QueryDescription`][QueryDescription]. This trait is
-used during cycle errors to give a "human readable" name for the query,
-so that we can summarize what was happening when the cycle occurred.
-Implementing this trait is optional if the query key is `DefId`, but
-if you *don't* implement it, you get a pretty generic error ("processing `foo`...").
-You can put new impls into the `config` module. They look something like this:
+There is an additional trait with more methods called
+[`self::config::QueryDescription`][QueryDescription]. This trait contains a few
+extra methods which are used by the query system.
```

Comment:
I think this is a bad place to document the query internals. These change semifrequently (you mentioned wanting to make this part of QueryConfig) and they aren't needed to understand how to use queries. I'd rather spend this space explaining local vs extern providers or something like that. Query internals should be documented in either query_impl or query_system IMO.

Reply:
Sounds good, I'll delete it. As for local vs extern providers I would first need to fully understand it myself before I could write about it so I guess that's follow up work :D

Reply:
Ok, opened https://github.com/rust-lang/rust/issues/104011 for documenting internals.